pipeline/pipeline.go

```diff
@@ -132,312 +113,211 @@ func (p *Pipeline) isOutputModule(name string) bool {
+			p.reqCtx.logger.Info("sending snapshot", zap.Strings("modules", p.reqCtx.Request().InitialStoreSnapshotForModules))
+			if err := p.sendSnapshots(p.reqCtx.Request().InitialStoreSnapshotForModules); err != nil {
 				return fmt.Errorf("send initial snapshots: %w", err)
 			}
 		}
 	}
 
-	p.initStoreSaveBoundary()
-
-	err = p.buildWASM(ctx, p.request, p.modules)
-	if err != nil {
+	if err = p.buildWASM(p.reqCtx.Context(), p.reqCtx.Request(), modules); err != nil {
 		return fmt.Errorf("initiating module output caches: %w", err)
 	}
 
-	for _, cache := range p.moduleOutputCache.OutputCaches {
-		atBlock := outputs.ComputeStartBlock(p.requestedStartBlockNum, p.outputCacheSaveBlockInterval)
-		if _, err := cache.LoadAtBlock(ctx, atBlock); err != nil {
-			return fmt.Errorf("loading outputs caches")
-		}
-	}
-
 	return nil
 }
 
-func (p *Pipeline) initStoreSaveBoundary() {
-	p.nextStoreSaveBoundary = p.computeNextStoreSaveBoundary(p.requestedStartBlockNum)
-}
+func (p *Pipeline) FlushStores(blockNum uint64) error {
+	storeFilters := map[string]bool{}
+	if p.isSubrequest {
+		// if subRequest we only want to flush output stores
+		storeFilters = p.outputModuleMap
 
-func (p *Pipeline) HandleStoreSaveBoundaries(ctx context.Context, span ttrace.Span, blockNum uint64) error {
-	for p.nextStoreSaveBoundary <= blockNum {
-		p.logger.Debug("saving stores on boundary", zap.Uint64("block_num", p.nextStoreSaveBoundary))
-		span.AddEvent("store_save_boundary_reach")
-		if err := p.saveStoresSnapshots(ctx, p.nextStoreSaveBoundary); err != nil {
-			if span != nil {
-				span.SetStatus(codes.Error, err.Error())
-			}
-			return fmt.Errorf("saving stores: %w", err)
-		}
-		p.bumpStoreSaveBoundary()
-		if isStopBlockReached(blockNum, p.request.StopBlockNum) {
-			break
-		}
 	}
-	return nil
-}
+	ranges, err := p.storeAccessor.Flush(p.reqCtx.Context(), blockNum, p.reqCtx.StopBlockNum(), storeFilters)
+	if err != nil {
+		return fmt.Errorf("saving stores: %w", err)
+	}
+	if p.isSubrequest {
+		// if we are in subrequest mode we want to update the partials written
+		// since we added a storeFilter for only ouput modules in subrequest mode, we are
+		// guaranteed that the ranges here are only for ouput modules
+		p.partialsWritten = append(p.partialsWritten, ranges...)
+		p.reqCtx.logger.Debug("adding partials written",
+			zap.Int("range_count", len(ranges)),
+			zap.Stringer("ranges", p.partialsWritten),
+			zap.Uint64("at_block_num", blockNum),
+		)
 
-func (p *Pipeline) bumpStoreSaveBoundary() bool {
-	p.nextStoreSaveBoundary = p.computeNextStoreSaveBoundary(p.nextStoreSaveBoundary)
-	return p.request.StopBlockNum == p.nextStoreSaveBoundary
-}
-func (p *Pipeline) computeNextStoreSaveBoundary(fromBlock uint64) uint64 {
-	nextBoundary := fromBlock - fromBlock%p.storeSaveInterval + p.storeSaveInterval
-	if p.isSubrequest && p.request.StopBlockNum != 0 && p.request.StopBlockNum < nextBoundary {
-		return p.request.StopBlockNum
```

Comment:
ouch I see those two methods removed, and some stuff in the Accessor that seems to do the same, although I remember needing to test these two methods with great care, to avoid a bunch of issues that I cannot recall.

What I see here is that some of those tests have simply been removed and not moved elsewhere. 
(searching `subreq, reqStart off bounds, current one off bound` doesn't show this test moved).

I probably didn't dig enough, but I'm a bit worried.